### ISSUE TITLE
Cubian-update does not work properly

### ISSUE THREAD

**Issue by sachilles** [cubieplayer/Cubian#329]
Hi,

after upgrading cubian via `cubian-update --verbose` the kernel 3.4.79_1 was downloaded from the git repository but **not** updated in the nand. I'm using a Cubieboard 1 with an installation into Nand. (Someone else pointed already to this fact; see also https://github.com/cubieplayer/Cubian/issues/294 .)

Best regards (and thanks for your work)!

**Comment by oguska** [comment]
Same issue here, cubian-update doesn't wor
k for nand.

cubie@Cubian:~$ sudo cubian-update
**\* Performing self-update
**\* Cubian updater v1.3.6 http://cubian.org
**\* Checking updates from server
**\* Downloading Firmware/Updates (this may take a while)
**\* Updating firmware
**\* Updating kernel
**\* Updating modules
**\* Depmod 3.4.79-sun4i
ERROR: could not open directory //lib/modules/3.4.79-sun4i: No such file or directory
FATAL: could not search modules: No such file or directory
**\* Updating fs
**\* Update success, A reboot is needed for changes to take effects

**Comment by michalliu** [comment]
How about `cubian-update -c` to clean the repository and rerun `cubian-update`

**Comment by sachilles** [comment]
Hi,

unfortunately this does not work. Below there is some of the latest output after cleaning the repository and applying the updates again.

```
*** Updating firmware
Using tag a10-3.4.79_1 which is latest in local repository
Note: checking out 'tags/a10-3.4.79_1'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b new_branch_name

HEAD is now at ba189c9... update kernel for A10
*** Updating kernel
*** Updating modules
*** Depmod 3.4.79-sun4i
ERROR: could not open directory //lib/modules/3.4.79-sun4i: No such file or directory
FATAL: could not search modules: No such file or directory
*** Updating fs
*** Checking updates from server
Checking updates from server https://github.com/cubieplayer/cubian-updates.git
Remote revision: a10-3.4.79_1
Local revision: a10-3.4.79_1
*** Your firmware is up to date
```

Best regards

**Comment by SimplyNaOH** [comment]
I had the same problem. Upon inspection I realized the script simply failed to copy the files in /root/.cubian-updates/. Manually copying the modules, boot files, and file-system and then executing "cubian-update -r" solved the issue.

Step by step instructions: (assuming you already enabled the root user by issuing "sudo passwd root")

``` bash
cubie@Cubian:~$ su
root@Cubian:~# cd /root/.cubian-updates/
root@Cubian:~/.cubian-updates# cp -d boot/uImage* /boot
root@Cubian:~/.cubian-updates# cp -d boot/config* /boot             
root@Cubian:~/.cubian-updates# cp -d boot/System* /boot
root@Cubian:~/.cubian-updates# cp -dR modules/* /lib/modules/
root@Cubian:~/.cubian-updates# cp -dR fs/* /
```

I had an issue with /lib/firmware/ap6210 so I deleted it and copied it again:

``` bash
root@Cubian:~/.cubian-updates# rm -r /lib/firmware/ap6210/
root@Cubian:~/.cubian-updates# cp -dR fs/* /
```

And finally:

``` bash
cubie@Cubian:~$ sudo cubian-update -r
```

**Comment by sachilles** [comment]
Hi,

indeed manually copying and `sudo cubian-update -r` did the job. Anyhow I guess one should inspect the `cubian-update` script if it fails so rigorously.

Thanks for your help.

**Comment by oguska** [comment]
Thanks mate

**Comment by cubieplayer** [comment]
Closed duplicated with #341
